web/packages/extension/webpack.config.js

```diff
@@ -42,12 +42,31 @@ module.exports = (env, argv) => {
                         to: "..",
                         transform(content) {
                             const manifest = JSON.parse(content.toString());
-                            const { version } = require("./package.json");
-                            Object.assign(manifest, { version });
+
+                            const packageVersion =
+                                process.env.npm_package_version;
+                            const versionChannel =
+                                process.env.CFG_RELEASE_CHANNEL || "nightly";
+                            const buildDate = new Date()
+                                .toISOString()
+                                .substr(0, 10);
```

Comment:
Per [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr), `substr` is deprecated, and `substring` is preferred.